aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -973,72 +314,18 @@ Tensor replication_pad2d_backward_cpu(
 }
 
 TORCH_IMPL_FUNC(replication_pad3d_out_cpu) (
-  const Tensor& input_, IntArrayRef paddingSize, const Tensor& output
+  const Tensor& input, IntArrayRef paddingSize, const Tensor& output
 ) {
-  int64_t pleft = paddingSize[0];
-  int64_t pright = paddingSize[1];
-  int64_t ptop = paddingSize[2];
-  int64_t pbottom = paddingSize[3];
-  int64_t pfront = paddingSize[4];
-  int64_t pback = paddingSize[5];
-  int64_t dimw = 3;
-  int64_t dimh = 2;
-  int64_t dimd = 1;
-  int64_t dimslices = 0;
-  int64_t nbatch = 1;
-
-  /* get contiguous input */
-  auto input = input_.contiguous();
-
-  if (input.dim() == 5) {
-    nbatch = input.size(0);
-    dimw++;
-    dimh++;
-    dimd++;
-    dimslices++;
-  }
-
-  /* sizes */
-  int64_t nslices = input.size(dimslices);
-  int64_t idepth  = input.size(dimd);
-  int64_t iheight = input.size(dimh);
-  int64_t iwidth  = input.size(dimw);
-  int64_t odepth  = output.size(dimd);
-  int64_t oheight = output.size(dimh);
-  int64_t owidth  = output.size(dimw);
-
-  /* resize output */
-  if (input.dim() == 4) {
-    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(input.scalar_type(), "replication_pad3d_cpu", [&] {
-      auto input_data = input.data_ptr<scalar_t>();
-      auto output_data = output.data_ptr<scalar_t>();
-      replication_pad3d_out_frame<scalar_t>(
-        input_data, output_data, nslices, iwidth, iheight, idepth,
-        owidth, oheight, odepth, pleft, pright, ptop, pbottom, pfront,
-        pback);
-      }
-    );
-  }
-  else
-  {
-    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(input.scalar_type(), "replication_pad3d_cpu", [&] {
-      auto input_data = input.data_ptr<scalar_t>();
-      auto output_data = output.data_ptr<scalar_t>();
-      replication_pad3d_out_batch<scalar_t>(
-        input_data, output_data, nslices, iwidth, iheight, idepth,
-        owidth, oheight, odepth, pleft, pright, ptop, pbottom, pfront,
-        pback,
-        nbatch);
-      }
-    );
-  }
+  replication_pad3d_kernel(kCPU, output, input, paddingSize);
 }
 
 Tensor& replication_pad3d_backward_out_cpu(const Tensor& gradOutput,
     const Tensor& input,
     IntArrayRef paddingSize,
     Tensor& gradInput)
 {
+  gradInput.resize_as_(input);
```

Comment:
Why is this change needed? Did the previous PR in this stack introduce a bag that is now being fixed in this one?

Reply:
This has a signature using `grad_input` as an alias:
```
replication_pad3d_backward.grad_input(Tensor grad_output, Tensor self, SymInt[6] padding, *, Tensor(a!) grad_input) -> Tensor(a!)
```

Originally, the reflection padding has this `grad_input.resize_as_(input)` but replication padding doesn't have this.
The resize logic is needed. IF the user manage to call the `replication_pad3d_backward(grad_output, self, padding, grad_input)` and `grad_input` is given with a wrong shape, it breaks. This should be a **very weird usage** but i think it should be better to add a resize logic when `grad_input` is given as an alias.

Reply:
Ok, thank you for double checking and the explanation!